rails/extensions/request_id_ext.rb

```diff
@@ -5,17 +5,22 @@
 #
 # See
 # ActionDispatch::RequestId
-
 module Daylight::RequestIdExt
+  def call(env)
+    req = ActionDispatch::Request.new env
+    if env["HTTP_X_REQUEST_ID"].presence
+      req.request_id = x_make_request_id(req.x_request_id)
+    else
+      req.request_id = make_request_id(req.x_request_id)
```

Comment:
why is this making a req only to set one attribute and then use that attribute later -- why make this req object at all?

Reply:
IIRC It's a monkey patch on ActionDispatch::RequestID, so Daylight is setting up some headers the way it likes it.

Reply:
But you're not using req here at all, except as a temporary holder for request_id.

Reply:
@app.call(env).tap { |_status, headers, _body| headers[ActionDispatch::RequestId::X_REQUEST_ID] = req.request_id }

I'm using it here

Reply:
right -- so why set it on req in the first place, why do you need the request model?